architecture/wm_sca/Readme.md

```diff
@@ -0,0 +1,36 @@
+<!---
+Copyright (C) 2015-2021, Wazuh Inc.
+Created by Wazuh, Inc. <info@wazuh.com>.
+This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+-->
+
+# Wazuh module: SCA architecture
+## Index
+1. [Purpose](#purpose)
+2. [Sequence Diagram](#sequence-diagram)
+
+## Purpose
+The security configuration assessment module (**SCA**) performs hardening and configuration scans based on custom policies, created and maintained by Wazuh's team based on CIS benchmarks.
+
+This module is composed of a main thread, from which two other secondary threads diverge. These threads are:
+
+- The main thread: In charge of parsing and converting to **JSON** the **YML** policies. These policies are then stored and executed accordingly. This thread contains high amounts of memory management and should be looked at closely, especially the `cis_db_for_hash` global structure, which seems to contain detailed information from all checks.
+
+- `wm_sca_dump_db_thread`: In charge of dumping the scan results into the manager's **SCA** database.
+
+- `wm_sca_request_thread`: In charge of requesting information if missing data is detected. Non-suspicious memory management.
```

Comment:
- In charge of processing dump requests from the manager side when synchronization fails.
- Remove `Non-suspicious memory management.`